src/back/space/src/api/create.cpp

```diff
@@ -0,0 +1,91 @@
+#include "create.hpp"
+#include "../service/service.hpp"
+#include "../../../shared/headers.hpp"
+#include <regex>
+
+namespace svetit::space::handlers {
+
+Create::Create(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerJsonBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+formats::json::Value Create::HandleRequestJsonThrow(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body,
+	server::request::RequestContext&) const
+{
+	formats::json::ValueBuilder res;
+
+	// todo - need to validate that this is valid uuid? need to check that user exists?
+	const auto& userId = req.GetHeader(headers::kUserId);
+	if (userId.empty()) {
+		res["err"] = "Empty userId header";
+		req.SetResponseStatus(server::http::HttpStatus::kUnauthorized);
+		return res.ExtractValue();
+	}
+
+	if (!body.HasMember("key")) {
+		LOG_WARNING() << "No key param in body";
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		res["err"] = "No key param in body";
+		return res.ExtractValue();
+	}
+
+	if (!body.HasMember("name")) {
+		LOG_WARNING() << "No name param in body";
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		res["err"] = "No name param in body";
+		return res.ExtractValue();
+	}
+
+	if (!body.HasMember("requestsAllowed")) {
+		LOG_WARNING() << "No requestsAllowed param in body";
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		res["err"] = "No requestsAllowed param in body";
+		return res.ExtractValue();
+	}
+
+	auto name = body["name"].ConvertTo<std::string>();
+	auto key = body["key"].ConvertTo<std::string>();
+	auto requestsAllowed = body["requestsAllowed"].ConvertTo<bool>();
+
+	boost::trim(name);
+	boost::trim(key);
```

Comment:
Нужно стараться там где это возможно использовать сериализацию/десериализацию 
```suggestion
	auto space = body.As<Space>();
```

Reply:
Реализовал базово именно в этом хендлере, на который ты ссылался в комменте (https://github.com/svetit-io/svetit/pull/38/commits/8da89968414f0831682df65fecfa73822a26abb2). Подумаю ещё дальше, как правильнее доработать.

Reply:
Готово (реализовал везде, где можно было парсить тело запроса).